elastic/kibana
# Fix sample data for share-capable objects

Resolves #116261.

To the Core team who owns most of the files that were changed, I recommend reviewing on a per-commit basis, see each commit message for details!

Manual testing steps:

1. Create two spaces
2. Go into one space and install sample data
3. Go into another space and install the same set of sample data
4. In the second space, observe that:
   - The saved object IDs are randomized
   - App links in the Home page work as expected
   - Uninstalling the sample data works as expected
